src/append/rolling_file/policy/compound/roll/fixed_window.rs

```diff
@@ -239,6 +246,119 @@ impl Deserialize for FixedWindowRollerDeserializer {
+        pub fn new(log_dir: Option<&Path>) -> Result<RollLockEx> {
+            let mut lock_path = PathBuf::new();
+            lock_path.push(log_dir.unwrap_or(Path::new("/tmp")));
+            lock_path.push("roll.lock");
+            let lock_file = File::create(&lock_path)?;
+            let ret = unsafe { libc::flock(lock_file.as_raw_fd(), libc::LOCK_EX | libc::LOCK_NB) };
+            if ret < 0 {
+                Err(Error::last_os_error())
+            } else {
+                Ok(RollLockEx { _lock_file: lock_file, lock_path })
+            }
+        }
+    }
+
+    impl Drop for RollLockEx {
+        fn drop(&mut self) {
+            remove_file(&self.lock_path).ok();
+        }
+    }
+}
+
+#[cfg(target_os = "solaris")]
+mod imp {
+    use std::fs::File;
+    use std::fs::remove_file;
+    use std::path::PathBuf;
+    use std::io::{Error, Result};
+    use std::os::unix::io::AsRawFd;
+    use std::path::Path;
+
+    pub struct RollLockEx {
+        _lock_file: File,
+        lock_path: PathBuf,
+    }
+
+    impl RollLockEx {
+        pub fn new(log_dir: Option<&Path>) -> Result<RollLockEx> {
+            let mut lock_path = PathBuf::new();
+            lock_path.push(log_dir.unwrap_or(Path::new("/tmp")));
+            lock_path.push("roll.lock");
+            let lock_file = File::create(&lock_path)?;
+            let flag = libc::LOCK_EX | libc::LOCK_NB;
+
+            let mut fl = libc::flock {
+                l_whence: 0,
+                l_start: 0,
+                l_len: 0,
+                l_type: 0,
+                l_pad: [0; 4],
+                l_pid: 0,
+                l_sysid: 0,
+            };
+
+            let (cmd, operation) = match flag & libc::LOCK_NB {
+                0 => (libc::F_SETLKW, flag),
+                _ => (libc::F_SETLK, flag & !libc::LOCK_NB),
+            };
+
+            match operation {
+                libc::LOCK_SH => fl.l_type |= libc::F_RDLCK,
+                libc::LOCK_EX => fl.l_type |= libc::F_WRLCK,
+                libc::LOCK_UN => fl.l_type |= libc::F_UNLCK,
+                _ => return Err(Error::from_raw_os_error(libc::EINVAL)),
+            }
+
+            let ret = unsafe { libc::fcntl(lock_file.as_raw_fd(), cmd, &fl) };
+            match ret {
+                -1 => match Error::last_os_error().raw_os_error() {
+                    Some(libc::EACCES) => return Err(Error::from_raw_os_error(libc::EWOULDBLOCK)),
+                    _ => return Err(Error::last_os_error())
+                },
+                _ => Ok(RollLockEx { _lock_file: lock_file, lock_path })
+            }
+        }
+    }
+
+    impl Drop for RollLockEx {
+        fn drop(&mut self) {
+            remove_file(&self.lock_path).ok();
+        }
+    }
+}
+
+#[cfg(windows)]
+mod imp {
+    use std::path::Path;
+    use std::io::Result;
+
+
+    pub struct RollLockEx {}
+
+    impl RollLockEx {
+        pub fn new(_log_dir: Option<&Path>) -> Result<RollLockEx> {
+            Ok(RollLockEx{})
```

Comment:
Yes - To request this gets merged with main log4rs repo

No - to get us going because AFAIK, we don't have a windows multiprocess agent.   @jens-tcell ?